server/src/internalClusterTest/java/org/elasticsearch/threadpool/SimpleThreadPoolIT.java

```diff
@@ -147,30 +154,36 @@ public void testThreadPoolMetrics() throws Exception {
         plugin.collect();
         ArrayList<String> registeredMetrics = plugin.getRegisteredMetrics(InstrumentType.LONG_GAUGE);
         registeredMetrics.addAll(plugin.getRegisteredMetrics(InstrumentType.LONG_ASYNC_COUNTER));
+
         tps.forEach(stats -> {
-            Map<String, Long> threadPoolMetrics = Map.of(
-                ThreadPool.THREAD_POOL_METRIC_NAME_COMPLETED,
-                stats.completed(),
-                ThreadPool.THREAD_POOL_METRIC_NAME_ACTIVE,
-                (long) stats.active(),
-                ThreadPool.THREAD_POOL_METRIC_NAME_CURRENT,
-                (long) stats.threads(),
-                ThreadPool.THREAD_POOL_METRIC_NAME_LARGEST,
-                (long) stats.largest(),
-                ThreadPool.THREAD_POOL_METRIC_NAME_QUEUE,
-                (long) stats.queue()
-            );
-            threadPoolMetrics.forEach((suffix, value) -> {
-                String metricName = ThreadPool.THREAD_POOL_METRIC_PREFIX + stats.name() + suffix;
-                List<Measurement> measurements;
-                if (suffix.equals(ThreadPool.THREAD_POOL_METRIC_NAME_COMPLETED)) {
-                    measurements = plugin.getLongAsyncCounterMeasurement(metricName);
-                } else {
-                    measurements = plugin.getLongGaugeMeasurement(metricName);
-                }
+            Map<String, Long> threadPoolStats = List.of(
```

Comment:
Nit: why not Map.ofEntries directly?

Reply:
This creates a sorted map. That's just for convenience to have a consistent order when logging the expect and actual result further below.

Reply:
Oh right, my eyes skipped over the "sorted" :) makes sense!